src/languageService/kustoLanguageService.ts

```diff
@@ -431,56 +431,37 @@ class KustoLanguageService implements LanguageService {
     doRangeFormat(document: ls.TextDocument, range: ls.Range): Promise<ls.TextEdit[]> {
         const rangeStartOffset: number = document.offsetAt(range.start);
         const rangeEndOffset: number = document.offsetAt(range.end);
-        const commands = this.getCommandsInDocument(document).then((commands) => {
-            const commandsInRange = commands
-                .map((command) => {
-                    // Chose only selected text from command.
-                    const commandSelectionStart =
-                        Math.max(rangeStartOffset, command.absoluteStart) - command.absoluteStart;
-                    const commandSelectionEnd = Math.min(rangeEndOffset, command.absoluteEnd) - command.absoluteStart;
-
-                    command.text =
-                        commandSelectionStart > commandSelectionEnd
-                            ? ''
-                            : command.text.substring(commandSelectionStart, commandSelectionEnd);
-
-                    return command;
-                })
-                .filter((command) => command.text.trim() != '');
-
-            const formattedCommands: string[] = commandsInRange.map((command) =>
-                Kusto.Data.Common.CslQueryParser.PrettifyQuery(command.text, '')
-            );
-            const formattedText: string = formattedCommands.join('\r\n\r\n');
+        const commands = this.getFormattedCommandsInDocumentV2(document, rangeStartOffset, rangeEndOffset);
 
-            return [ls.TextEdit.replace(range, formattedText)];
-        });
+        if (!commands.rangeBeforeFormat || commands.formattedCommands.length === 0) {
```

Comment:
maybe better to rename rangeBeforeFormat to originalRange because i assumed it means the range of characters that come before the formatted range.